src/cmap/connection_requester.rs

```diff
@@ -31,112 +26,87 @@ pub(super) fn channel(
 #[derive(Clone, Debug)]
 pub(super) struct ConnectionRequester {
     address: StreamAddress,
-    sender: mpsc::UnboundedSender<oneshot::Sender<RequestedConnection>>,
+    sender: mpsc::UnboundedSender<oneshot::Sender<ConnectionRequestResult>>,
     handle: PoolWorkerHandle,
 }
 
 impl ConnectionRequester {
```

Comment:
The error handling was split/duplicated between the `ConnectionRequester` and `ConnectionPool::check_out`, and it was pretty confusing. To simplify this, I moved most of the logic to `check_out` and only kept the parts in here that involved communicating with the worker thread. Once a response was received, it is returned and then processed for errors and what not by `check_out`.